### DISCUSSION TITLE
Draft Mode & Revalidation

### DISCUSSION THREAD

**Discussion by adrocodes** [vercel/next.js#55692] [App Router]
I am working on a client project and we're having some issues with `draftMode` not pulling in the latest content. I _feel_ like it was working fine until recently, maybe before the new `13.5.x` releases.

I haven't worked on a replication repo which is why I don't have a issue, I am hoping someone has run into a similar issue and has a fix or suggestion.

I have a page that is similar to this;

```tsx
async function getData() {
  const { data } = await getClient(isPreview()).query({...})
  return data
}

export default async function Homepage() {
  const data = await getData()
  return <>...</>
}

export const revalidate = 43200; // 12 hours in seconds
```

In this case `getClient` will return a `ApolloClient` instance and `isPreview()` essentially wraps `draftMode().isEnabled`. The `ApolloClient` instance will change based on the `isPreview()` value. We're using Dato so we're adding the required headers.

I've added `fetchPolicy: isPreview() ? 'no-cache' : undefined` to the `query` options and that seemed to work (until recently).

I also changed my `getClient` method to just return a `fetch` wrapper similar to what Dato outlines in their [docs](https://www.datocms.com/docs/next-js/optimizing-calls-with-react-cache-function). That also didn't make a difference.

I don't really want to use the experimental apollo package for app directory, but since `fetch` also didn't work, I can't imagine it would make a huge difference.

---

From Dato CMS there is a Preview Link which runs the `draftMode().enable()` and then redirects to the page.

I'd expect when I jump into `draftMode` that the page would pull in the latest content, but it doesn't. To get the preview to show the latest content I either need to run a `revalidatePath` any time there is a draft change or rebuild the entire site which obviously isn't ideal.

In my mind, I want this page to be built at deploy time and revalidate every 12 hours. When I jump into `draftMode` the revalidate time should be ignored and the page should pull in the latest content on every refresh.

Keen to hear if anyone else is facing a similar issue or if I am doing something wrong.

**Comment by wpiron10** [comment]
@adrocodes hi, did you get any workaround / solution ? i have the same problem
